### PR TITLE
Gracefully handle the potential for missing attributes

### PR DESCRIPTION
**Pull request by gep13**
As described in GH-515

### CODE REVIEW

File: src/Cake.Common/Tools/DupFinder/DupFinderRunner.cs [hunk 1]
```diff
@@ -195,7 +195,7 @@ private void AnalyzeResultsFile(FilePath resultsFilePath, bool throwOnDuplicates
 
             foreach (var duplicate in duplicates)
             {
-                _log.Warning("Duplicate Located with a cost of {0}, across {1} Fragments", duplicate.Attribute("Cost"), duplicate.Descendants("Fragment").Count());
+                _log.Warning("Duplicate Located with a cost of {0}, across {1} Fragments", duplicate.Attribute("Cost").Value, duplicate.Descendants("Fragment").Count());
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/DupFinder/DupFinderRunner.cs]: What happens if the attribute is null here?

Comment (gep13) [inline src/Cake.Common/Tools/DupFinder/DupFinderRunner.cs]: The Unit Tests fail :smile: But I forgot to run them before pushing :smile:

**Comment by gep13** [comment]
@patriksvensson just pushed another set of changes.

**Comment by patriksvensson** [comment]
Looks good to me :+1:

**Comment by gep13** [comment]
@patriksvensson I have kicked the Travis build, in the hope that it will work this time.  If so, can you merge? Thanks!

**Comment by patriksvensson** [comment]
@gep13 It failed again so I restarted it. Will merge after CI finish.

**Comment by gep13** [comment]
@patriksvensson woot! This finally passed, can you merge? Thanks!

**Comment by patriksvensson** [comment]
@gep13 Merged :+1:

**Pull request merged**